### PR TITLE
[WFCORE-5526] Remove Legacy Security Realms from WildFly Core and WildFly.

### PR DESCRIPTION
**Pull request by darranl**
https://issues.redhat.com/browse/WFCORE-5526

### CODE REVIEW

File: security/WFCORE-5526_Remove_Legacy_Security_Realms.adoc [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [x] Engineering
+
+* [x] QE
+
+As a removal RFE this does not require new feature test development, however both teams will need
+to review existing testsuites for tests dependent on the Vault integration.  Identified tests will
```
Comment (OndrejKotek) [inline security/WFCORE-5526_Remove_Legacy_Security_Realms.adoc:50]: Vault integration -> legacy security realms

and the like.

File: security/WFCORE-5526_Remove_Legacy_Security_Realms.adoc [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+this just updates the feature packs to remove the use of security realms but at this point tests
+may still be defining their own security realms.
+
+A second issue https://issues.redhat.com/browse/WFLY-15067[WFLY-15067] will prepare WildFly for the
+security realms becoming unavailable in an upcomming WildFly Core release.
+
+https://issues.redhat.com/browse/WFCORE-5526[WFCORE-5526] will then proceed to completely remove
+the security realms from WildFly Core.
+
+A final task https://issues.redhat.com/browse/WFLY-15067[WFLY-15068] will complete the removal in
```
Comment (OndrejKotek) [inline security/WFCORE-5526_Remove_Legacy_Security_Realms.adoc:168]: WFLY-15067 -> WFLY-15068

File: security/WFCORE-5526_Remove_Legacy_Security_Realms.adoc [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+This task will include a pass through of all existing community documentation and remove all
+references to the security realms.
+
+A migration "article" will be added to the documentation, this article will identify where
+security-realm integration has been removed with some high level examples of alternatives.
+The article will contain some simple examples but a complete like for like comparison would be
+impractical.
+
+At the end of all of the removals we will come back to the issue 
+https://issues.redhat.com/browse/WFLY-15067[WFLY-15068] to perform a final clean up and refactor
```
Comment (OndrejKotek) [inline security/WFCORE-5526_Remove_Legacy_Security_Realms.adoc:253]: WFLY-15067 -> WFLY-15068

File: security/WFCORE-5526_Remove_Legacy_Security_Realms.adoc [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+N/A
+
+=== Non-Requirements
+
+Later enhancements will be looking at how we can evolve the default configuration further, this enhancement
+is specifcially covering the removal.
+
+The following issues have been raised for follow up activities to be considered later:
+
+ * ????
```
Comment (OndrejKotek) [inline security/WFCORE-5526_Remove_Legacy_Security_Realms.adoc:150]: https://issues.redhat.com/browse/WFCORE-5549

File: security/WFCORE-5526_Remove_Legacy_Security_Realms.adoc [hunk 5]
```diff
@@ -0,0 +1,379 @@
...
+|HTTP Authentication
+|application-http-authentication
+
+|ApplicationRealm
+|SASL Authentication
+|application-sasl-authentication
+
+|ApplicationRealm
+|SSL
+|applicationSSC
```
Comment (mchoma) [inline security/WFCORE-5526_Remove_Legacy_Security_Realms.adoc:121]: When I render this analysis document [1], there is some problem with formatting of table? I guess`|===` is missing here?

[1] https://github.com/wildfly/wildfly-proposals/blob/8152bca32b8ec6af85e693b7979d2f54a7bbc22c/security/WFCORE-5526_Remove_Legacy_Security_Realms.adoc

File: security/WFCORE-5526_Remove_Legacy_Security_Realms.adoc [hunk 6]
```diff
@@ -452,6 +452,90 @@ The following table identifies the tests in WildFly Core and WildFly affected by
 |Evolved https://issues.redhat.com/browse/WFLY-15057[WFLY-15057]
 |===
 
+|WildFly
```
Comment (mchoma) [inline security/WFCORE-5526_Remove_Legacy_Security_Realms.adoc:455]: When I look in preview mode https://github.com/wildfly/wildfly-proposals/blob/c581b74477897429aff919444aa51675a5c8bc0f/security/WFCORE-5526_Remove_Legacy_Security_Realms.adoc This table is not formatted correctly.

File: security/WFCORE-5526_Remove_Legacy_Security_Realms.adoc [hunk 7]
```diff
@@ -0,0 +1,576 @@
...
+=== Issue
+
+* https://issues.redhat.com/browse/WFCORE-5526[WFCORE-5526 - Remove legacy security realm support]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1095[EAP7-1095 - Remove the legacy SecurityRealms]
+* https://issues.redhat.com/browse/WFLY-15001[WFLY-15001 - Update feature packs to remove use of legacy security realms]
+* https://issues.redhat.com/browse/WFLY-15067[WFLY-15067 - Prepare for removal of legacy security realm support]
+* https://issues.redhat.com/browse/WFLY-15067[WFLY-15068 - Final stage for the removal of legacy security realms.]
```
Comment (OndrejKotek) [inline security/WFCORE-5526_Remove_Legacy_Security_Realms.adoc:32]: WFLY-15067 -> WFLY-15068

**Comment by darranl** [comment]
@OndrejKotek the requested changes should have been applied now, the latest version with the tables rendering correctly can be seen at:
    https://github.com/wildfly/wildfly-proposals/blob/95dc8ef8ed99d2f2d9b59426a90e0a187e00dec5/security/WFCORE-5526_Remove_Legacy_Security_Realms.adoc

**Comment by OndrejKotek** [comment]
Thank you @darranl, I have no more suggestions. QE ack for this analysis.

**Pull request merged**